lib/srv/alpnproxy/auth/auth_proxy_test.go

```diff
@@ -0,0 +1,56 @@
+package alpnproxyauth
+
+import (
+	"context"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/stretchr/testify/require"
+	"net"
+	"testing"
+)
+
+type mockAuthGetter struct {
+	servers []types.Server
+}
+
+func (m mockAuthGetter) GetClusterName(...services.MarshalOption) (types.ClusterName, error) {
+	return nil, nil
+}
+
+func (m mockAuthGetter) GetAuthServers() ([]types.Server, error) {
+	return m.servers, nil
+}
+
+func TestDialLocalAuthServerNoServers(t *testing.T) {
+	s := NewAuthProxyDialerService(nil, mockAuthGetter{servers: []types.Server{}})
+	_, err := s.dialLocalAuthServer(context.Background())
+	require.Error(t, err)
+	require.Equal(t, "empty auth servers list", err.Error())
+}
+
+func TestDialLocalAuthServerNoAvailableServers(t *testing.T) {
+	server1, err := types.NewServer("s1", "auth", types.ServerSpecV2{Addr: "invalid:8000"})
+	require.Nil(t, err)
+	s := NewAuthProxyDialerService(nil, mockAuthGetter{servers: []types.Server{server1}})
+	_, err = s.dialLocalAuthServer(context.Background())
+	require.Error(t, err)
+	require.Equal(t, "all auth servers unavailable: invalid:8000: dial tcp: lookup invalid: no such host", err.Error())
```

Comment:
CI disagrees with this exact error; a bit of a moot point if you might change the error anyway, but I'd be a little lax on the exact error message - something along the lines of `require.Contains(t, "all auth servers unavailable: invalid:8000", err.Error())` would be better.

Reply:
+1